File: page_objects/contact_page.py
from page_objects.base_page import BasePage


class ContactPage(BasePage):
    def fill_form(self, name: str, email: str, subject: str, message: str):
        self.page.fill('#name', name)
        self.page.fill('#email', email)
        self.page.fill('#subject', subject)
        self.page.fill('#message', message)

    def submit_form(self):
        self.page.click('button[type="submit"]')

    def get_success_message(self):
        return self.page.text_content('.alert-success')

    def get_error_message(self):
        return self.page.text_content('.alert-danger')

File: pageObjects/ContactPage.ts

import { Page } from 'playwright';

export class ContactPage {
  constructor(private page: Page) {}

  async fillForm(formData: { name: string; email: string; subject: string; message: string }) {
    await this.page.fill('#name', formData.name);
    await this.page.fill('#email', formData.email);
    await this.page.fill('#subject', formData.subject);
    await this.page.fill('#message', formData.message);
  }

  async submitForm() {
    await this.page.click('button[type="submit"]');
  }

  async getSuccessMessage() {
    return this.page.textContent('.alert-success');
  }

  async getErrorMessage() {
    return this.page.textContent('.alert-danger');
  }
}
